src/index.ts

```diff
@@ -1,13 +1,11 @@
-import * as extend from 'extend';
-import * as _request from 'request';  // for types only
-import * as request from 'retry-request';
+import axios, {AxiosError, AxiosRequestConfig, AxiosResponse} from 'axios';
 
 const BASE_URL = 'http://metadata.google.internal/computeMetadata/v1';
 
-export type Options = _request.CoreOptions&{property?: string, uri?: string};
+export type Options = AxiosRequestConfig&{property?: string, uri?: string};
 
 export type Callback =
-    (error: NodeJS.ErrnoException|null, response?: _request.RequestResponse,
+    (error: NodeJS.ErrnoException|null, response?: AxiosResponse<string>,
      metadataProp?: string) => void;
 
 export function _buildMetadataAccessor(type: string) {
```

Comment:
Given that we are making a semver major change anyway; can we switch this to an awaitable (thenable-returning) function? Maybe we should also throw if the user passed an argument that is `typeof 'function'`. That will help people migrate to the new version.